datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1517,9 +1517,13 @@ fn compare_join_arrays(
             },
             DataType::Date32 => compare_value!(Date32Array),
             DataType::Date64 => compare_value!(Date64Array),
-            _ => {
+            dt => {
                 return not_impl_err!(
-                    "Unsupported data type in sort merge join comparator"
+                    "{}",
```

Comment:
IIRC `not_impl_err!` works like `format!`, so you don't need to create an intermediate string here (apart from the fact that the nesting is not that readable.

Same goes for the other change.

Reply:
Ah, got it. Fixed.